experimental/devices/mfrc522/mfrc522.go

```diff
@@ -0,0 +1,732 @@
+		case <-time.After(100 * time.Millisecond):
+			// do nothing
+		}
+	}
+}
+
+// AntiColl performs the collision check for different cards.
+func (r *Dev) AntiColl() ([]byte, error) {
+
+	if err := r.devWrite(commands.BitFramingReg, 0x00); err != nil {
+		return nil, err
+	}
+
+	backData, _, err := r.CardWrite(commands.PCD_TRANSCEIVE, []byte{commands.PICC_ANTICOLL, 0x20}[:])
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(backData) != 5 {
+		return nil, fmt.Errorf("Back data expected 5, actual %d", len(backData))
+	}
+
+	crc := byte(0)
+
+	for _, v := range backData[:4] {
+		crc = crc ^ v
+	}
+
+	if crc != backData[4] {
+		return nil, errors.New(fmt.Sprintf("CRC mismatch, expected %02x actual %02x", crc, backData[4]))
+	}
+
+	return backData, nil
+}
+
+// CRC calculates the CRC of the data using the card chip.
+func (r *Dev) CRC(inData []byte) ([]byte, error) {
+	if err := r.clearBitmask(commands.DivIrqReg, 0x04); err != nil {
+		return nil, err
+	}
+	if err := r.setBitmask(commands.FIFOLevelReg, 0x80); err != nil {
+		return nil, err
+	}
+	for _, v := range inData {
+		if err := r.devWrite(commands.FIFODataReg, v); err != nil {
+			return nil, err
+		}
+	}
+	if err := r.devWrite(commands.CommandReg, commands.PCD_CALCCRC); err != nil {
+		return nil, err
+	}
+	for i := byte(0xFF); i > 0; i-- {
+		n, err := r.devRead(commands.DivIrqReg)
+		if err != nil {
+			return nil, err
+		}
+		if n&0x04 > 0 {
+			break
+		}
+	}
+	lsb, err := r.devRead(commands.CRCResultRegL)
+	if err != nil {
+		return nil, err
+	}
+
+	msb, err := r.devRead(commands.CRCResultRegM)
+	if err != nil {
+		return nil, err
+	}
+	return []byte{lsb, msb}, nil
+}
+
+// SelectTag selects the FOB device by device UUID.
+func (r *Dev) SelectTag(serial []byte) (byte, error) {
+	dataBuf := make([]byte, len(serial)+2)
+	dataBuf[0] = commands.PICC_SElECTTAG
+	dataBuf[1] = 0x70
+	copy(dataBuf[2:], serial)
+	crc, err := r.CRC(dataBuf)
+	if err != nil {
+		return 0, err
+	}
+	dataBuf = append(dataBuf, crc[0], crc[1])
+	backData, backLen, err := r.CardWrite(commands.PCD_TRANSCEIVE, dataBuf)
+	if err != nil {
+		return 0, err
+	}
+
+	var blocks byte
+
+	if backLen == 0x18 {
+		blocks = backData[0]
+	} else {
+		blocks = 0
+	}
+	return blocks, nil
+}
+
+// AuthStatus indicates the authentication response, could be one of AuthOk, AuthReadFailure or AuthFailure
+type AuthStatus byte
+
+const (
+	AuthOk AuthStatus = iota
+	AuthReadFailure
+	AuthFailure
+)
+
+func (r *Dev) auth(mode byte, blockAddress byte, sectorKey []byte, serial []byte) (AuthStatus, error) {
+	buffer := make([]byte, 2)
+	buffer[0] = mode
+	buffer[1] = blockAddress
+	buffer = append(buffer, sectorKey...)
+	buffer = append(buffer, serial[:4]...)
+	_, _, err := r.CardWrite(commands.PCD_AUTHENT, buffer)
+	if err != nil {
+		return AuthReadFailure, err
+	}
+	if n, err := r.devRead(commands.Status2Reg); err != nil || n&0x08 == 0 {
+		return AuthFailure, err
+	}
+	return AuthOk, nil
+}
+
+// StopCrypto stops the crypto chip.
```

Comment:
Does it need to be exported?

Reply:
I think yes, for some use-cases. Or we can move the refinement of the public API to the next iteration.